src/AWSS3.jl

```diff
@@ -202,19 +202,11 @@ Is there an object in `bucket` at `path`?
 function s3_exists(
     aws::AbstractAWSConfig, bucket, path; version::AbstractS3Version=nothing, kwargs...
 )
-    @repeat 2 try
-        s3_get_meta(aws, bucket, path; version=version, kwargs...)
-
-        return true
-
-    catch e
-        @delay_retry if ecode(e) in ["NoSuchBucket", "404", "NoSuchKey", "AccessDenied"]
-        end
-
-        @ignore if ecode(e) in ["404", "NoSuchKey", "AccessDenied"]
-            return false
-        end
-    end
+    l = s3_list_objects(aws, bucket, path; kwargs...)
```

Comment:
Isn't this a more expensive operation though? Maybe at least set `max_items=1`?

Reply:
I've changed this to `max_items=1`.  Still haven't done anything like a thorough analysis of how the performance is.